### ISSUE TITLE
Unexpected "no such binding" errors when aggregate used in subquery of SELECT VALUE from source

### ISSUE THREAD

**Issue by therapon** [partiql/partiql-lang-kotlin#212]
The following error doesn't make any sense to me whatsoever since `foo` is the alias being given to the result of the nested query.

```
partiql> select value foo.cnt from (select count(*) as cnt from [1, 2, 3]) as foo
   | 
software.amazon.sql.eval.EvaluationException: No such binding: foo
        Evaluator Error: at line 1, column 14: Binding does not exist, <UNKNOWN> : <UNKNOWN>

        at software.amazon.sql.impl.eval.EvaluatingCompiler$compileVariableReference$$inlined$thunkEnv$1.invoke(Thunk.kt:237)
        at software.amazon.sql.impl.eval.EvaluatingCompiler$compileVariableReference$$inlined$thunkEnv$1.invoke(Thunk.kt)
        at software.amazon.sql.impl.eval.EvaluatingCompiler$compilePath$$inlined$thunkEnv$1.invoke(Thunk.kt:237)
        at software.amazon.sql.impl.eval.EvaluatingCompiler$compilePath$$inlined$thunkEnv$1.invoke(Thunk.kt)
        at software.amazon.sql.impl.eval.EvaluatingCompiler$compileSelect$1$$special$$inlined$thunkEnvValue$1.invoke(Thunk.kt:237)
        at software.amazon.sql.impl.eval.EvaluatingCompiler$compileSelect$1$$special$$inlined$thunkEnvValue$1.invoke(Thunk.kt)
        at software.amazon.sql.impl.eval.EvaluatingCompiler$compileSelect$1$1$getQueryThunk$$inlined$thunkEnv$2.invoke(Thunk.kt:258)
        at software.amazon.sql.impl.eval.EvaluatingCompiler$compileSelect$1$1$getQueryThunk$$inlined$thunkEnv$2.invoke(Thunk.kt)
        at software.amazon.sql.impl.eval.EvaluatingCompiler$compileSelect$1$1$getQueryThunk$$inlined$thunkEnv$4.invoke(Thunk.kt:237)
        at software.amazon.sql.impl.eval.EvaluatingCompiler$compileSelect$1$1$getQueryThunk$$inlined$thunkEnv$4.invoke(Thunk.kt)
        at software.amazon.sql.impl.eval.EvaluatingCompiler$compile$1.eval(EvaluatingCompiler.kt:205)
        at software.amazon.sqlcli.Repl$run$totalMs$1.invoke(Repl.kt:92)
        at software.amazon.sqlcli.Repl$run$totalMs$1.invoke(Repl.kt:35)
        at software.amazon.sqlcli.StopwatchKt.timeIt(stopwatch.kt:21)
        at software.amazon.sqlcli.Repl.run(Repl.kt:70)
        at software.amazon.sqlcli.Main.runRepl(main.kt:158)
        at software.amazon.sqlcli.Main.main(main.kt:143)

ERROR! (5 ms)
```

The result I expect is: `<< 3 >>`

This also seems wrong--in this case, `baz` is the alias assigned to the single from source within the nested query.

```
partiql> select value foo.cnt from (select count(baz.bar) as cnt from [ { 'bar': 1 }, { 'bar': 2}] as baz) as foo
   | 
software.amazon.sql.eval.EvaluationException: No such binding: baz
        Evaluator Error: at line 1, column 41: Binding does not exist, <UNKNOWN> : <UNKNOWN>

        at software.amazon.sql.impl.eval.EvaluatingCompiler$compileVariableReference$$inlined$thunkEnv$1.invoke(Thunk.kt:237)
        at software.amazon.sql.impl.eval.EvaluatingCompiler$compileVariableReference$$inlined$thunkEnv$1.invoke(Thunk.kt)
        at software.amazon.sql.impl.eval.EvaluatingCompiler$compilePath$$inlined$thunkEnv$1.invoke(Thunk.kt:237)
        at software.amazon.sql.impl.eval.EvaluatingCompiler$compilePath$$inlined$thunkEnv$1.invoke(Thunk.kt)
        at software.amazon.sql.impl.eval.EvaluatingCompiler$compileSelect$1$1$getQueryThunk$$inlined$thunkEnv$2.invoke(Thunk.kt:251)
        at software.amazon.sql.impl.eval.EvaluatingCompiler$compileSelect$1$1$getQueryThunk$$inlined$thunkEnv$2.invoke(Thunk.kt)
        at software.amazon.sql.impl.eval.EvaluatingCompiler$compileSelect$1$1$getQueryThunk$$inlined$thunkEnv$4.invoke(Thunk.kt:237)
        at software.amazon.sql.impl.eval.EvaluatingCompiler$compileSelect$1$1$getQueryThunk$$inlined$thunkEnv$4.invoke(Thunk.kt)
        at software.amazon.sql.impl.eval.EvaluatingCompiler$compile$1.eval(EvaluatingCompiler.kt:205)
        at software.amazon.sqlcli.Repl$run$totalMs$1.invoke(Repl.kt:92)
        at software.amazon.sqlcli.Repl$run$totalMs$1.invoke(Repl.kt:35)
        at software.amazon.sqlcli.StopwatchKt.timeIt(stopwatch.kt:21)
        at software.amazon.sqlcli.Repl.run(Repl.kt:70)
        at software.amazon.sqlcli.Main.runRepl(main.kt:158)
        at software.amazon.sqlcli.Main.main(main.kt:143)

ERROR! (5 ms)
```

Changing `count(baz.bar) as cnt` to simply `baz.bar as cnt` yields the following (which is the expected result), so probably this has something to do with aggregate support.

```
partiql> select value foo.cnt from (select baz.bar as cnt from [ { 'bar': 1 }, { 'bar': 2}] as baz) as foo
   | 
===' 
1
2
--- 
Result type was BAG and contained 2 items
OK! (6 ms)
```

**Comment by Jun0413** [comment]
I ran into this same issue. It seems that if a subquery has aggregation, there will be a binding problem. Is this an expected behavior?

**Comment by alancai98** [comment]
@Jun0413 this is not the expected behavior.

The issue comes from one of the rewriters incorrectly handling subqueries with aggregations. Which eventually led to a wrong evaluation path taken in [`EvaluatingCompiler`](https://github.com/partiql/partiql-lang-kotlin/blob/de1fe32b1d535a9d56b3337b1de9c10780422745/lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt#L972) 

This issue should be fixed in #344.

**Comment by alancai98** [comment]
Fixed by #344. Merged into master by #356.
